### PR TITLE
feature(sf): feature sf alert desc

### PR DESCRIPTION
**Pull request by Q16G**

### CODE REVIEW

**Review by VillanCh — APPROVED** [review]

File: common/yak/yakurl/ssaurl_test.go [hunk 1]
```diff
@@ -118,13 +118,13 @@ func TestSFURl(t *testing.T) {
 		res, err := SendURL(local, progID, "/", `
 		target2(* #-> as $a) 
 		target1() as $target1
-		alert $target1 for "alert information"
+		alert $target1 for {msg: 'alert information'}
```
Comment (wlingze) [inline common/yak/yakurl/ssaurl_test.go:121]: add bool field in `ExtraInfo`, this `alert $a for "msg"` just get "msg" not `{"Msg": "msg", ...}`.
this Field `OnlyMsg` can set in compiler(syntaxflow antlr visitor)(visit_statement.go:177)

File: common/yak/ssaapi/sf_result.go [hunk 2]
```diff
@@ -61,17 +64,23 @@ func (r *SyntaxFlowResult) Name() string {
...
 }
 
-func (r *SyntaxFlowResult) GetAlertMsg(name string) (string, bool) {
+func (r *SyntaxFlowResult) GetAlertInfo(name string) (*sfvm.ExtraDescInfo, bool) {
 	if r == nil {
-		return "", false
+		return nil, false
 	}
 
 	if r.memResult != nil {
```
Comment (wlingze) [inline common/yak/ssaapi/sf_result.go:72]: use `r.rule` in all time, and set `r.rule` when create SyntaxFlowResult. 
in this pr modify function: `CreateResultFromQuery`

File: common/syntaxflow/sfvm/vm.go [hunk 3]
```diff
@@ -92,18 +96,27 @@ func (s *SyntaxFlowVirtualMachine) Compile(text string) (frame *SFFrame, ret err
...
-	frame.AllowIncluded = result.allowIncluded
-	frame.Purpose = result.purpose
-	frame.Language = result.language
-	frame.Severity = result.severity
+	frame.rule.Content = text
+	frame.rule.Title = result.title
+	frame.rule.TitleZh = result.title_zh
+	frame.rule.Description = result.description
+	frame.rule.Type = schema.ValidRuleType(result.purpose)
+	frame.rule.AllowIncluded = func() bool {
```
Comment (wlingze) [inline common/syntaxflow/sfvm/vm.go:108]: simplify this code. jist `rule.AllowInclude = result.allowInclude!=""` and `if rule.AllowInclude{}`

File: common/schema/syntaxflow_rule.go [hunk 4]
```diff
@@ -88,7 +88,7 @@ type SyntaxFlowRule struct {
 	TitleZh     string
 	Description string
 	Tag         string
-
+	AlertDesc   string
```
Comment (wlingze) [inline common/schema/syntaxflow_rule.go:91]: just save Struct type, then can use this in alltime, and just marshal/unmarshal once for one instance.

File: common/syntaxflow/sfvm/visitor.go [hunk 5]
```diff
@@ -16,6 +16,7 @@ import (
 type SyntaxFlowVisitor struct {
```
Comment (wlingze) [inline common/syntaxflow/sfvm/visitor.go:16]: this function should simplify? just embed `schema.SyntaxflowRule`, like `Frame`?

Comment (wlingze) [reply common/syntaxflow/sfvm/visitor.go:16]: implement this modify in pr: #1895

**Review by wlingze — APPROVED** [review]

**Pull request merged**